resolwe/flow/executors/run.py

```diff
@@ -0,0 +1,283 @@
+                            spawn_processors.append(obj)
+                    elif line.strip().startswith('export'):
+                        file_name = line[6:].strip()
+
+                        export_folder = settings.FLOW_EXECUTOR['UPLOAD_DIR']
+                        unique_name = 'export_{}'.format(uuid.uuid4().hex)
+                        export_path = os.path.join(export_folder, unique_name)
+
+                        self.exported_files_mapper[self.data_id][file_name] = unique_name
+
+                        shutil.move(file_name, export_path)
+                    else:
+                        # If JSON, save to MongoDB
+                        updates = {}
+                        for obj in iterjson(line):
+                            for key, val in six.iteritems(obj):
+                                if key.startswith('proc.'):
+                                    if key == 'proc.error':
+                                        process_error.append(val)
+                                        if not process_rc:
+                                            process_rc = 1
+                                            updates['process_rc'] = process_rc
+                                        updates['process_error'] = process_error
+                                        updates['status'] = settings.Data_status['STATUS_ERROR']
+                                    elif key == 'proc.warning':
+                                        process_warning.append(val)
+                                        updates['process_warning'] = process_warning
+                                    elif key == 'proc.info':
+                                        process_info.append(val)
+                                        updates['process_info'] = process_info
+                                    elif key == 'proc.rc':
+                                        process_rc = int(val)
+                                        updates['process_rc'] = process_rc
+                                        if process_rc != 0:
+                                            updates['status'] = settings.Data_status['STATUS_ERROR']
+                                    elif key == 'proc.progress':
+                                        process_progress = int(float(val) * 100)
+                                        updates['process_progress'] = process_progress
+                                else:
+                                    output[key] = val
+                                    updates['output'] = output
+
+                        if updates:
+                            updates['modified'] = now().isoformat()
+                            self.update_data_status(**updates)
+
+                        if process_rc > 0:
+                            log_file.close()
+                            json_file.close()
+                            os.chdir(CWD)
+                            self._send_manager_command('finish', process_rc=process_rc)
```

Comment:
As discussed, protocol commands should not be passed as strings around the place.